core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -841,16 +974,17 @@ private[kafka] class Processor(val id: Int,
     Option(metrics.metric(metrics.metricName("io-wait-ratio", MetricsGroup, metricTags))).fold(0.0)(m =>
       Math.min(m.metricValue.asInstanceOf[Double], 1.0))
   },
+    //TODO is this an issue if we change the metrics tags?
```

Comment:
I think so because it is not backward compatible. It seems that we already kept that metrics for backward compatibility reason. I am not sure if it was deprecated at some point or not. If it would, we could perhaps drop it in a future release. Is this the only problematic metric?

Reply:
I've pushed an update to keep all existing metric names.